### PR TITLE
Add intro services template

### PR DESCRIPTION
**Pull request by ericl**
Adds an intro services template

@scottsun94 , after this merges could you also add it to the templates file with a similar config as the Intro to Workspaces template?

@edoakes , let me know your feedback. I'm not a services expert though so ideally someone on the team could take a pass afterwards to improve the notebook.

### CODE REVIEW

**Comment by scottsun94** [comment]
LGTM. Do we land on "serve deploy" or "serve publish"?

File: templates/intro-services/README.ipynb
Comment (akshay-anyscale) [inline templates/intro-services/README.ipynb:184]: We should add a note here about using the serve logger as well (and in the code example).

**Comment by edoakes** [comment]
> Adds an intro services template
> 
> @scottsun94 , after this merges could you also add it to the templates file with a similar config as the Intro to Workspaces template?
> 
> @edoakes , let me know your feedback. I'm not a services expert though so ideally someone on the team could take a pass afterwards to improve the notebook.

Will do a quick review now and happy to do another pass through it after it's merged.

File: templates/intro-services/README.ipynb
Comment (edoakes) [inline templates/intro-services/README.ipynb:25]: nit: to match the convention on other examples we have elsewhere
```suggestion
    " To get started, create a file called `main.py` and fill it with the following skeleton code:\n",
```

File: templates/intro-services/README.ipynb
Comment (edoakes) [inline templates/intro-services/README.ipynb:42]: (same as above)
```suggestion
    "my_app = FastAPIDeployment.bind()"
```

File: templates/intro-services/README.ipynb
Comment (edoakes) [inline templates/intro-services/README.ipynb:37]:
```suggestion
    "    # FastAPI will automatically parse the HTTP request for us.\n",
    "    # Because "name" is specified as an argument, it'll be parsed from the query params of the request.\n",
```

Comment (edoakes) [reply templates/intro-services/README.ipynb:37]: Maybe we should link out to fastapi docs?

File: templates/intro-services/README.ipynb
Comment (edoakes) [reply templates/intro-services/README.ipynb:184]: Let's have Gene add this once the logging UI is available?

**Review by edoakes — APPROVED** [review]
Looks good only nits, I can iterate on it once `serve deploy` is in.

I'd like to have content somewhere about how to pass arguments to an application because it's a common use case, especially to have different dev. vs. prod args, and it's best to suggest an opinionated way to do it. Do you think a section here makes sense? Or should we have a separate "advanced" notebook? Don't want to have notebook explosion...

File: templates/intro-services/README.ipynb
Comment (ericl) [reply templates/intro-services/README.ipynb:184]: I can add an example.

@akshay-anyscale can you explain to me what I should note here? If I use `print()`, will that not show up in the logs at all?

File: templates/intro-services/README.ipynb
Comment (ericl) [reply templates/intro-services/README.ipynb:91]: Changed to lower case (except when referring to it as "Anyscale Service"), since all upper case would be pretty obnoxious.

File: templates/intro-services/README.ipynb
Comment (ericl) [reply templates/intro-services/README.ipynb:184]: Ok maybe I'll just merge this for now to unblock things.

**Comment by ericl** [comment]
> I'd like to have content somewhere about how to pass arguments to an application because it's a common use case, especially to have different dev. vs. prod args, and it's best to suggest an opinionated way to do it.

Sounds good. Can you let me know some snippets (don't actually know what you have in mind to show)?

File: templates/intro-services/README.ipynb
Comment (akshay-anyscale) [reply templates/intro-services/README.ipynb]: it will show up but it won't be annotated with metadata like request id, etc. or follow any formatting (if we want to use json formatting). I agree we can merge and ask Gene to update this

**Comment by edoakes** [comment]
> > I'd like to have content somewhere about how to pass arguments to an application because it's a common use case, especially to have different dev. vs. prod args, and it's best to suggest an opinionated way to do it.
> 
> Sounds good. Can you let me know some snippets (don't actually know what you have in mind to show)?

https://docs.ray.io/en/latest/serve/advanced-guides/app-builder-guide.html#pass-arguments-to-applications

**Pull request merged**
